### PR TITLE
Refine CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by thorn0**
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Preview](https://github.com/thorn0/prettier/blob/contributing-md/CONTRIBUTING.md)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Can you also simply explain that `run_spec` function accept object as first argument.

**Comment by fisker** [comment]
One more thing, please metion that `babel-ts` parser will be tested when `typescript` appears, and when `babel` parser is tested in `tests/js/` `espree` and `meriyah` will be tested too.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,27 +4,74 @@ To get up and running, install the dependencies and run the tests:
...
+
+```js
+run_spec(__dirname, ["babel"], { trailingComma: "es5" });
+```
+
+Signature:
+
+```ts
+function run_spec(
+  fixtures: { dirname: string; snippets?: string[] } | string,
```
Comment (fisker) [inline CONTRIBUTING.md:32]: `snippets` support object with properties `name`(test name), `filename`, `code`, `output`(direct assert output string)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,27 +4,63 @@ To get up and running, install the dependencies and run the tests:
...
+## Debugging
+
+To debug Prettier locally, you can either debug it in Node (recommended) or the browser.
+
+- The easiest way to debug it in Node is to create a local test file with some example code you want formatted and either run it in an editor like VS Code or run it directly via `./bin/prettier.js <your_test_file>`.
+- The easiest way to debug it in the browser is to build Prettier's website locally (see [`website/README.md`](website/README.md)).
+
+## Other
+
+Run `yarn fix` to lint and format all files.
```
Comment (fisker) [inline CONTRIBUTING.md:65]: I think  better suggest use `yarn lint`, if there is bug  in those eslint rules,        they may lost their work.

Comment (thorn0) [reply CONTRIBUTING.md:65]: If you're not sure about some rules, let's better disable them. Many people (me included) use autofix on save. I don't want to lose work too.

Comment (fisker) [reply CONTRIBUTING.md:67]: I think enabled rules don't have known issues, in editor if you found something not right you can undo, but if editor already closed and haven't committed, run `yarn fix` you can't find your original code back.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -4,30 +4,76 @@ To get up and running, install the dependencies and run the tests:
...
+- **`parsers`**: A list of parser names. The tests verify that the parsers in this list produce the same output. If the list includes `typescript`, then `babel-ts` is included implicitly. If the list includes `babel`, and the current directory is inside `tests/js`, then `espree` and `meriyah` are included implicitly.
+- **`options`**: In addition to Prettier's formatting option, can contain the `errors` property to specify that it's expected that the formatting shouldn't be successful and an error should be thrown for all (`errors: true`) or some combinations of input entries and parsers.
+
+The implementation of `run_spec` can be found in `tests_config/run_spec.js`.
```
Comment (fisker) [inline CONTRIBUTING.md:54]: Link to `tests_config/run_spec.js` file.

File: CONTRIBUTING.md (hunk 2 above)
Comment (thorn0) [reply CONTRIBUTING.md:67]: I just might not notice the editor did that.

**Review by fisker — APPROVED** [review]
Good job.

**Pull request merged**